lib/cartopy/mpl/gridliner.py

```diff
@@ -156,6 +191,25 @@ def __init__(self, axes, crs, draw_labels=False, xlocator=None,
         #: Whether to draw labels on the right hand side of the map.
         self.ylabels_right = draw_labels
 
+        if not auto:
+            #: Whether to draw x labels inline
+            self.x_inline = x_inline
+            #: Whether to draw y labels inline
+            self.y_inline = y_inline
+        else:
+            if isinstance(self.axes.projection, _SUPPORTED_PROJS):
```

Comment:
I think this block could be replaced by:
```python
self.x_inline = not isinstance(self.axes.projection, _SUPPORTED_PROJS) and isinstance(self.axes.projection, _X_INLINE_PROJS)
self.y_inline = not isinstance(self.axes.projection, _SUPPORTED_PROJS) and isinstance(self.axes.projection, _Y_INLINE_PROJS)
```

Reply:
I don't think it handles the else statement; also I think it's much more readable and intuitive for person reading code if I spread it out like this (personally I had a hard time understanding the one liners)